src/storage/secondary/column.rs

```diff
@@ -57,22 +57,25 @@ pub trait ColumnBuilder<A: Array> {
 }
 
 /// Iterator on a column. This iterator may request data from disk while iterating.
-#[async_trait]
 pub trait ColumnIterator<A: Array> {
+    type NextFuture<'a>: Future<Output = StorageResult<Option<(u32, A)>>> + 'a
+    where
+        Self: 'a;
+
     /// Get a batch and the starting row id from the column. A `None` return value means that
     /// there are no more elements from the block. By using `expected_size`, developers can
     /// get an array of NO MORE THAN the `expected_size` on supported column types.
-    async fn next_batch(
-        &mut self,
+    fn next_batch<'a>(
```

Comment:
```suggestion
    fn next_batch(
```

No need to explicitly write `'a` here.

In the below parameters, simply use `&mut self`, `Option<&BitVec>`, and `Self::NextFuture<'_>`.

Reply:
Sorry, I'm a beginner of rust. I tried this suggestion, but compiler said:
```
error[E0623]: lifetime mismatch
   --> src/storage/secondary/column/concrete_column_iterator.rs:298:10
    |
297 |         filter_bitmap: Option<&BitVec>,
    |                               ------- this parameter and the return type are declared with different lifetimes...
298 |     ) -> Self::NextFuture<'_> {
    |          ^^^^^^^^^^^^^^^^^^^^
    |          |
    |          ...but data from `filter_bitmap` is returned here
```
Is there any way to fix that?

Reply:
hmmm... Looks like a compiler bug. No need to fix.